client.go

```diff
@@ -163,14 +163,16 @@ func (c *client) Scan(s *hrpc.Scan) ([]*hrpc.Result, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		scanres = res.(*pb.ScanResponse)
 		results = append(results, scanres.Results...)
 
 		// TODO: The more_results field of the ScanResponse object was always
 		// true, so we should figure out if there's a better way to know when
 		// to move on to the next region than making an extra request and
 		// seeing if there were no results
-		for len(scanres.Results) != 0 {
+		for len(scanres.Results) != 0 &&
+				(numberOfRows == 0 || uint32(len(results)) < numberOfRows) {
```

Comment:
`numberOfRows == 0` should probably be checked before even sending the first request. Also, the case of "if there are results and numberOfRows == 0" doesn't really make sense, so maybe just drop it?